week02/Number of Matching Subsequences.cpp

```diff
@@ -0,0 +1,25 @@
+//TC : O(n + m * mxlen(m) * log(n)) where O(n) for building map + O(no.of words * max len of word * finding position with upper_bound) 
+//SC : O(257)

```

Comment:
Wrong. The space complexity here is `O(n)`. You are storing all the indices in `freq` eventually.